### PR TITLE
Implement math intrinsics for WGSL

### PR DESCRIPTION
**Pull request by jkwak-work**
This commit implements math related intrinsics and a few others for WGSL.

The implementation is based on the following doc,
 https://www.w3.org/TR/WGSL

### CODE REVIEW

**Review by csyonghe — APPROVED** [review]
Looks good!

File: tests/wgsl/math.slang [hunk 1]
```diff
@@ -0,0 +1,285 @@
+//TEST:SIMPLE(filecheck=WGSL): -stage compute -entry computeMain -target wgsl
```
Comment (csyonghe) [inline tests/wgsl/math.slang:1]: Since we only have `WGSL:` comments in the test, I guess we don't need the other test lines here.

**Comment by jkwak-work** [comment]
Many CI actions are failing but they don't seem to be related to my change, because they seem to be failing to checkout from github repository.
The important one, "Windows Release", is passing, which is a good sign.

I will re-trigger the failed tests later and see if it works out.

File: tests/wgsl/math.slang (hunk 1 above)
Comment (jkwak-work) [reply tests/wgsl/math.slang:1]: I removed other targets.

**Comment by csyonghe** [comment]
@jkwak-work any idea on the failed CI runs?

**Comment by jkwak-work** [comment]
> @jkwak-work any idea on the failed CI runs?

I am still investigating.

**Comment by jkwak-work** [comment]
It turned out that the CI failures are due to my change.
It was hitting an assertion failure when I reproduced it on WSL debug build at
https://github.com/shader-slang/slang/blob/25d155937402120292cff9e6f5a31c1d45a2ecbf/tools/slang-test/slang-test-main.cpp#L957

The target, SLANG_WGSL, was not handled in this function.

**Comment by jkwak-work** [comment]
All CI are now passing.
Merging now.

**Pull request merged**